esp/bindings/SOAP/xpp/fxpp/FragmentedXmlPullParser.h

```diff
@@ -0,0 +1,2145 @@
+ * group members created by the group can inherit namespaces defined for the group.
+ */
+#define FXPP_PROPAGATE_NAMESPACES     XmlFragmentRules(0x04)
+
+/**
+ * Instruct the parser to require a non-empty unique identifier for a fragment. The identifier is
+ * used to detect circular references between fragments. Identifiers are required for all fragments
+ * with the exception of the initial parser input and embedded fragments.
+ *
+ * Some identifiers are generated automatically by fragment implementations, while others must be
+ * assigned manually. Fragments with manually assigned values must assign the same identifier to
+ * every occurrence of the same content. For example, two references to the same external fragment
+ * may produce two separate fragment instances but both instances must specify the same identifier.
+ */
+#define FXPP_REQUIRE_FRAGMENT_UID     XmlInternalFragmentRules(0x0100)
+
+/**
+ * Instruct the parser to enforce the injection restriction requiring an in-progress element.
+ */
+#define FXPP_REQUIRE_CURRENT_ELEMENT  XmlInternalFragmentRules(0x0200)
+
+/**
+ * Reserved element names used internally by the parser.
+ */
+namespace reserved
+{
+    constexpr static const char* SYNTHESIZED_ROOT = "synthesized_root";
+    constexpr static const char* SYNTHESIZED_GROUP = "synthesized_group";
+}
+
+/**
+ * `IXmlFragment` implementation that assumes the character pointers given in its constructor will
+ * remain valid for the life of the instance.
+ */
+class CSimpleXmlFragment : public CInterfaceOf<IXmlFragment>
+{
+public:
+    virtual const char*      queryUId() const override { return m_uid; }
+    virtual const char*      queryContent() const override { return m_content; }
+    virtual int              queryContentLength() const override { return m_length; }
+    virtual XmlFragmentRules queryRules() const override { return m_rules; }
+private:
+    const char*      m_uid;
+    const char*      m_content;
+    int              m_length;
+    XmlFragmentRules m_rules;
+public:
+    CSimpleXmlFragment(const char* uid, const char* content, int length, XmlFragmentRules rules)
+        : m_uid(uid)
+        , m_content(content)
+        , m_length(length)
+        , m_rules(rules)
+    {
+    }
+};
+
+/**
+ * `IFagmentedXmlPullParser` implementation that uses a stack of `XmlPullParser` instances to
+ * support parsing fragments within fragments. To a great extent, it is a wrapper of
+ * `XmlPullParser`, and what works with the original parser should work with this class with no
+ * required changes. There are exceptions:
+ * 
+ * - Mixed content, the ability for an element to contain both text and child element content, is
+ *   explicitly not supported. Attempting to enable this mode will result in an exception.
+ * - Whitespace content events are ignored by default.
+ */
+class CFragmentedXmlPullParser : public IFragmentedXmlPullParser
```

Comment:
Since we use interfaces it would be better to put these classes in *.cpp files and expose a create function so that externally only interfaces are used.  Interfaces would need to be declared outside of any classes.  Any classes that require the user to derive there own classes from them should go in *.ipp files.